actions/aaa_wrexmods_dependencies_MOD.js

```diff
@@ -21,123 +21,40 @@ WrexMODS.latest_changes = "The module installer WORKS NOW";
-			}
-
-			if(module.attempts >= this.MaxInstallAttemptsPerModule){
-				console.error("DBM MODS (Node Module Installer v2.5): Could not automatically install " + moduleName + ". \n\n (Install attempt limit reached) \n Please try to run your bot CMD/Terminal (Ctrl + Shift + Right Click [In Windows] To Open a CMD Window) and do 'node bot.js' at least once to allow the installer to run. \n If that still fails please do 'npm install --save " + moduleName + "' and restart your bot before you continue.");
-				module.errored = true;
-				reject(module);
-			}else{
-
-				try {			
-					console.log("DBM MODS (Node Module Installer v2.5) Attempting To Install Node Module: '" + moduleName + "'. Please wait...\n");	
-
-					const child = require('child_process');
-					let cliCommand = 'npm install ' + moduleName + " --loglevel=error " + (isGlobal ? "-g" : "--save");
-					child.execSync(cliCommand,{cwd: require('path').dirname(process.argv[1]),stdio:[0,1,2]});
-
-					try {
-						require.resolve(modulePath);
-						module.installed = true;
-
-						console.log("DBM MODS (Node Module Installer v2.5) Node Module '" + moduleName + "' has been Installed. You MAY need to restart your bot if theres errors.");	
-						resolve(module)	;	
-
-					} catch (error) {
-						console.error("DBM MODS (Node Module Installer v2.5): Node Module  '" + moduleName + "' failed to install. Attempt Number: " + module.attempts + " out of " + this.MaxInstallAttemptsPerModule);
-						if(module) WrexMODS.CheckAndInstallNodeModule(module.name);
-					}
-	
-				} catch (error) {
-
-					module.errored = true;
-
-					if(error.message.includes("Command failed") || error.message.includes("Not found")|| error.message.includes("Not Found")){
-						console.log("DBM MODS (Node Module Installer v2.5): Node Module  '" + moduleName + "' does not exist!");
-						console.error(error.message);
-					}else{
-						console.error("DBM MODS (Node Module Installer v2.5): MAIN ERROR. Report the information below to DBM Mods Support!");
-						console.dir(this.modules);
-						console.error(error.message);
-						console.log("----------------------------------------");
-					}
-					
-				}
-				
-			}		
+			resolve(require(moduleName));
+		} catch (e) {
+			console.log(`Failed to Install ${moduleName}, please re-try or install manually with "npm i ${moduleName}"`);
+			reject('Installer Failed.');
 		}
 	});
 }
 
 
 
 WrexMODS.require = function(moduleName){
-    
-	/// <summary> Custom require function that will attempt to install the module if it doesn't exist</summary>
-	/// <returns type="Object">The required module</returns>
-	
-	this.CheckAndInstallNodeModule(moduleName);	
-	
-	const botPath = path.dirname(process.argv[1]);
-	const modulePath = path.join(botPath, "node_modules", moduleName);
-
-	return require.main.require(modulePath);
+	try {
+		return require(moduleName);
+	} catch (e) {
+		this.CheckAndInstallNodeModule(moduleName).then(mod => {
+			return mod;
+		}).catch(err => {
```

Comment:
this won't do what you're expecting

Reply:
just do it like it's being done in the v12 branch.
https://github.com/dbm-network/mods/blob/0da5a5ae2745a262173b6dcea1b60149879cdbca/actions/aaa_dbmmods_dependencies_MOD.js#L19